com.unity.render-pipelines.universal/Runtime/RenderTargetBufferSystem.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UnityEngine.Rendering.Universal.Internal
+{
+    //NOTE: This class is meant to be removed when RTHandles get implemented in urp
+    internal sealed class RenderTargetBufferSystem
+    {
+        RenderTargetHandle RTA;
+        RenderTargetHandle RTB;
+        bool m_FirstIsBackBuffer = true;
+        RenderTextureDescriptor m_Desc;
+        FilterMode m_FilterMode;
+
+        int m_NameA;
+        int m_NameB;
+
+        public RenderTargetBufferSystem(string name)
+        {
+            m_NameA = Shader.PropertyToID(name + "A");
+            m_NameB = Shader.PropertyToID(name + "B");
+            RTA.Init(name + "A");
+            RTB.Init(name + "B");
+        }
+
+        public RenderTargetHandle GetBackBuffer()
+        {           
+            return m_FirstIsBackBuffer ? RTA : RTB;
+        }
+
+        public RenderTargetHandle GetBackBuffer(CommandBuffer cmd)
+        {
+            Initialize(cmd);
+            return m_FirstIsBackBuffer ? RTA : RTB;
+        }
+
+        public RenderTargetHandle GetFrontBuffer(CommandBuffer cmd)
+        {
+            Initialize(cmd);
+            return m_FirstIsBackBuffer ? RTB : RTA;
+        }
+
+        public void Swap()
+        {
+            m_FirstIsBackBuffer = !m_FirstIsBackBuffer;
+        }
+
+        void Initialize(CommandBuffer cmd)
+        {
+            cmd.GetTemporaryRT(m_NameA, m_Desc, m_FilterMode);
+            cmd.GetTemporaryRT(m_NameB, m_Desc, m_FilterMode);
```

Comment:
Where do we release second bufffer, I am guessing the first one is released with old logic in universal renderer?

Reply:
It doesn't release yet. 
So by default I think it should always release at the same point as the active color target.
But do we also want an option in the Swap() function (`Swap(bool release frontbuffer)`) so we can specify that we will no longer need a second buffer so we can get rid of the memory sooner?

Although we are never sure if the user will need the second buffer in their own passes so maybe it's not worth trying to get rid of it sooner.

Reply:
If we release after swapping we have to make sure to get if from pool upon access again. Is there any downside of always releasing instead of adding option?

Reply:
I am no expert on the workings of GetTemporaryRT but as I understand it. Releasing it every time should actually be fine since everything is pooled anyway. That way we get rid of it as soon as possible and it really doesn't cost anything extra if we need it 2 lines further in the code again. But like I said I am no expert on it so maybe someone can confirm this?

Reply:
Yes that is correct, we should release at the end of the frame or even sooner if possible. Because once it is released can be immedetiatelly re-used by other requests, which would create another one if not present